allennlp/modules/seq2seq_encoders/bidirectional_language_model_transformer.py

```diff
@@ -39,10 +39,8 @@ def attention(query: torch.Tensor,
 
 def subsequent_mask(size: int, device: str = 'cpu') -> torch.Tensor:
     """Mask out subsequent positions."""
-    attn_shape = (1, size, size)
-    mask = np.triu(np.ones(attn_shape), k=1).astype('uint8')
-    mask = (torch.from_numpy(mask) == 0)
-    return mask.to(device)
+    mask = torch.tril(torch.ones(size, size, device=device)).unsqueeze(0)
```

Comment:
Shouldn't the type of `uint8` be preserved?

Reply:
The old return type was bool because of `torch.from_numpy(mask) == 0`. At the only place this is being used in the code base currently we have `subsequent = subsequent_mask(timesteps, device).int()` so the return type doesn't matter. i could cast to `int` to match this use case if that's preferable.